cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs

```diff
@@ -232,26 +233,63 @@ instance ToJSON adr => ToJSON (NetworkTopology adr) where
 readTopologyFile :: ()
   => forall adr. FromJSON adr
   => NodeConfiguration -> CT.Tracer IO (StartupTrace blk) -> IO (Either Text (NetworkTopology adr))
-readTopologyFile NodeConfiguration{ncTopologyFile=TopologyFile topologyFilePath, ncConsensusMode} tracer = runExceptT $ do
+readTopologyFile NodeConfiguration{ncTopologyFile=TopologyFile topologyFilePath, ncConsensusMode, ncProtocolFiles} tracer = runExceptT $ do
   bs <- handleIOExceptionsLiftWith handler $ BS.readFile topologyFilePath
-  topology@RealNodeTopology{ntUseBootstrapPeers} <-
+  topology@RealNodeTopology{ntUseLedgerPeers, ntUseBootstrapPeers, ntPeerSnapshotPath} <-
     liftEither . first handlerJSON $
       eitherDecode $ LBS.fromStrict bs
 
   unless (isValidTrustedPeerConfiguration topology) $
     throwError handlerBootstrap
 
+  when (isBlockProducer && useLedgerPeers ntUseLedgerPeers) $
+    liftIO $ CT.traceWith tracer
+           $ NetworkConfigUpdateWarning
+           $ createMsg "Use of ledger peers is not recommended for BP operation"
+
+  when (isJust ntPeerSnapshotPath && not (useLedgerPeers ntUseLedgerPeers) && isBlockProducer) $
+    liftIO $ CT.traceWith tracer
+           $ NetworkConfigUpdateInfo
+           $ createMsg "Ledger peers and peer snapshot, although specified in the topology file, are disabled in line with recommended BP operation"
+
+  when (inPraosMode && isJust ntPeerSnapshotPath &&  not (useLedgerPeers ntUseLedgerPeers)) $
+    if isBlockProducer
+    then liftIO $ CT.traceWith tracer
+           $ NetworkConfigUpdateWarning
+           $ createMsg
+           $  "Peer snapshot file specified but topology disables ledger peers - ignoring file. "
+           <> "To turn off this message remove peerSnapshotFile from the topology file."
+    else liftIO $ CT.traceWith tracer
+           $ NetworkConfigUpdateWarning
+           $ createMsg
+           $  "Peer snapshot file specified but topology disables ledger peers - ignoring file. "
+           <> "To turn off this message enable the use of ledger peers or remove peerSnapshotFile from the topology file."
+
+
+  when (inGenesisMode && not (useLedgerPeers ntUseLedgerPeers) && not isBlockProducer) $
+    liftIO $ CT.traceWith tracer
+           $ NetworkConfigUpdateWarning
+           $ createMsg "It is recommended to use ledger peers and peer snapshot file for relay operations in Genesis mode"
```

Comment:
When in genesis mode and there, and ledger peers are disabled, the node should error (this message needs to be an error, and the node needs to exit - am I right that the node used to exit in this case?).

Ping @amesgen.